pkg/transformer/kubernetes/k8sutils.go

```diff
@@ -127,24 +127,38 @@ func PrintList(objects []runtime.Object, opt kobject.ConvertOptions) error {
 	f := transformer.CreateOutFile(opt.OutFile)
 	defer f.Close()
 
-	var err error
 	var files []string
 
 	// if asked to print to stdout or to put in single file
 	// we will create a list
 	if opt.ToStdout || f != nil {
 		list := &api.List{}
-		list.Items = objects
+		// convert objects to versioned and add them to list
+		for _, object := range objects {
+			versionedObject, err := convertToVersion(object, unversioned.GroupVersion{})
+			if err != nil {
+				return err
+			}
 
-		// version each object in the list
-		list.Items, err = convertToVersion(list.Items)
-		if err != nil {
-			return err
-		}
+			//make sure that Serices are first in the List
+			switch object.(type) {
+			case *api.Service:
```

Comment:
To simplify this switch you can also use `object.GetObjectKind().GroupVersionKind().Kind == "Service"` to do the checks.